test/framework/workload.go

```diff
@@ -123,7 +123,7 @@ func (w *WorkloadCluster) writeKubeconfigToDisk(ctx context.Context, secretName
 	if err := w.Provider.UpdateKubeConfig(&kubeconfig, w.ClusterName); err != nil {
 		return fmt.Errorf("failed to update kubeconfig for cluster: %s", err)
 	}
-	writer, err := filewriter.NewWriter(w.ClusterConfigFolder)
+	writer, err := filewriter.NewWriter(filepath.Dir(filePath))
```

Comment:
This was the assumption that broke everything
the output of `KubeconfigFilePath()` was being passed here as `filePath`
That output could have an arbitrary value depending on the cluster name. However, this code assumes the parent folder was `ClusterConfigFolder`.

But that was only true if `ClusterConfigFolder` == `ClusterName`. Which is not always true.